addons/l10n_in/views/res_company_view.xml

```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+
+    <record id="l10n_in_view_company_form" model="ir.ui.view">
+        <field name="name">l10n.in.view.company.form</field>
+        <field name="model">res.company</field>
+        <field name="priority" eval="100"/>
+        <field name="inherit_id" ref="base.view_company_form"/>
+        <field name="arch" type="xml">
+            <xpath expr="//field[@name='vat']" position="attributes">
+                <attribute name="string">GSTIN</attribute>
```

Comment:
overwrite fields_view_get() of res.company and res.partner to display the country_id.vat_label or 'TIN'